Central-Server/API/main.py

```diff
@@ -1,38 +1,151 @@
+args = parser.parse_args()
+
+# Initialize the API
+app = APIFlask(__name__, title="Kamaji", static_url_path="/static", static_folder="./static")
+app.register_blueprint(jobs_blueprint)
+app.register_blueprint(job_queue_blueprint)
+app.register_blueprint(perms_blueprint)
+CORS(app)
+
+# Initialize datastreamer connectiosn
+m_thread = BoardManagerThread()
+
+# Set up server configs
+app.config['SPEC_FORMAT'] = 'json'
+app.config['SERVERS'] = [
+    {
+        'name': 'Dev Server',
+        'url': 'http://localhost/api/'
+    }
+    # TODO: Add prod
+]
+
+# Set up API endpoints
+@app.route("/")
+def api_index():
+    """/api/ -- Returns literal 'OK'"""
+    return "OK"
+
+@app.errorhandler(403)
+def access_forbidden_error(e):
+    """err 403 -- Returns normal 403 error"""
+    if random.randint(0, 100000) == 42069: # We are a really serious space program, check out the CEO of SpaceX, we like the same jokes
+        return "<html><h1>Access forbidden :(</h1><img src=\"/api/static/image_480.png\"/></html>", 200
+    return jsonify(error=str(e)), 403
+
+def generate_jobs_table():
+    """Generates database if it doesn't exist"""
+    conn = database.connect()
+    cursor = conn.cursor()
+    cursor.execute("SELECT 1 FROM pg_tables WHERE tablename='hilsim_runs'")
+    exists = cursor.fetchone()
+    if not exists:
+        with open("database.sql") as f:
+            cursor.execute(f.read())
+        conn.commit()
+        conn.close()
+        cursor.close()
+
+@app.route('/boards', methods=["GET"])
+@app.output(internal.boards.BoardList())
+def list_boards():
+    """
+    /api/boards/ -- List all the boards that are connected to the server
+    """
+    board_list = []
+    for board in m_thread.threads:
+        board_list.append({
+            "id": board.board_id,
+            "is_ready": board.is_ready,
+            "job_running": board.job_running,
+            "board_type": board.board_type,
+            "running": board.running
+        })
+    return {
+        "boards": board_list
+    }, 200
+
+@app.route('/internal/queue', methods=["GET"])
+def list_internal_queue():
+    """
+    /api/internal/queue -- Debug function to index the size of the current queue
+    """
+    return jsonify({"size": len(m_thread.queue), "content": [str(e) for e in m_thread.queue]}), 200
+
+if __name__ == "__main__":
+    # Perform final initialization
+    print("Attempting to initialize server..")
+    jobs = []
+
+    # Start the running of manager thread
+    m_thread.start()
+
+    # Re-initialize database
+    generate_jobs_table()
+
+    port = int(os.environ.get('PORT', 443))
+    print("PORT:", port)
+    if(args.runmode=="dev"):
+        # Init dev server
+        print("Initialized development websocket server on ws://localhost:" + str(port))
+        socketio = SocketIO(app, cors_allowed_origins='*')
+        app.run(debug=True, host="0.0.0.0", port=port, use_reloader=False)
+        socketio.run()
+    else:

```

Comment:
check if `args.runmode` is `prod` here, instead of running on any non dev input

Reply:
Good idea, will add to notes

Reply:
Done.